randovania/games/samus_returns/exporter/patch_data_factory.py

```diff
@@ -58,18 +115,66 @@ def _start_point_ref_for(self, node: Node) -> dict:
         else:
             return {}
 
+    def _key_error_for_node(self, node: Node, err: KeyError) -> KeyError:
+        return KeyError(f"{self.game.region_list.node_name(node, with_region=True)} has no extra {err}")
+
+    def _level_name_for(self, node: Node) -> str:
+        region = self.game.region_list.nodes_to_region(node)
+        return region.extra["scenario_id"]
+
+    def _teleporter_ref_for(self, node: Node, actor_key: str = "actor_name") -> dict:
+        try:
+            return {
+                "scenario": self._level_name_for(node),
+                "actor": node.extra[actor_key],
+            }
+        except KeyError as e:
+            return {}
+            raise self._key_error_for_node(node, e)
+
     def _pickup_detail_for_target(self, detail: ExportedPickupDetails) -> dict | None:
+        alt_model = _ALTERNATIVE_MODELS.get(detail.model, [detail.model.name])
+        model_names = alt_model
+
+        if detail.other_player:
+            if model_names == ["offworld"]:
+                base_icon = "unknown"
+                model_names = ["itemsphere"]
```

Comment:
If you're already copying code related to multiworld, then you should have a rdvgame that multiworld to test